src/fiber/channel.rs

```diff
@@ -4983,6 +5009,40 @@ impl ChannelActorState {
         }
     }
 
+    fn get_active_tlcs_for_settlement(&self, for_remote: bool) -> Vec<SettlementTlc> {
+        let (mut received_tlcs, mut offered_tlcs) = (
+            self.get_active_received_tlcs(for_remote),
+            self.get_active_offered_tlcs(for_remote),
+        );
+        let (mut a, mut b) = if for_remote {
+            (received_tlcs, offered_tlcs)
+        } else {
+            for tlc in received_tlcs.iter_mut().chain(offered_tlcs.iter_mut()) {
+                // Need to flip these fields for the counterparty.
+                tlc.flip_mut();
+            }
+            (offered_tlcs, received_tlcs)
+        };
+        a.sort_by(|x, y| u64::from(x.tlc_id).cmp(&u64::from(y.tlc_id)));
+        b.sort_by(|x, y| u64::from(x.tlc_id).cmp(&u64::from(y.tlc_id)));
```

Comment:
This part of get tlcs logic is same with `get_active_tlcs`, maybe we should split current `get_active_tlcs` into two function, and `get_active_tlcs` is just filtering out tlcs we are interested in.

Reply:
let me create a new PR for refactoring, thanks.